common/user/session.ts

```diff
@@ -74,13 +74,13 @@ export async function updateSessionRolesOfUser(userID: string) {
 
 // O(n)
 // Currently used in session manager to log out all sessions created by a specific device token
-export async function deleteSessionsByDevice(deviceId: string | null, user: User) {
+export async function deleteSessionsByDevice(deviceId: string | null, user: User | undefined) {
     if (!deviceId) {
         return; // do nothing if deviceId is undefined
     }
     const sessionsToDelete = [];
     for await (const [sessionToken, sessionUser] of userSessions.iterator() as AsyncIterable<[string, GraphQLUser]>) {
-        if (sessionUser.deviceId === deviceId && sessionUser.userID === user.userID) {
+        if (sessionUser.deviceId === deviceId && sessionUser.userID === user?.userID) {
```

Comment:
Do we actually need the userID check here? I can't think of a scenario where we need this, and removing the sessionUser.userID === user?.userID check would also allow admins to log out other users.

Reply:
As DeviceIDs are generated on a user controlled device, malicious users could on purpose generate device ids that collide with other users.